### DISCUSSION TITLE
How to update a package to specific version?

### DISCUSSION THREAD

**Discussion by dayongxie** [pdm-project/pdm#3084] [Q&A] [answered]
As far as I know, I can use the `pdm update` command to update a package to the latest version.

 If I want to update to a specific version or revert to an earlier version, the only way I know is to change the pyproject.toml file. 

Are there any other ways to do this?

**Comment by frostming** [comment]
`pdm add` is the command to use whenever you want to modify the dependencies in `pyproject.toml`. It just does similar to edit and install. For example, `pdm add foo==1.0.0` overwrites the specification of `foo` package and restrict it to `==1.0.0`.

If you do not want to touch the `pyproject.toml` you can also run `pdm lock --exclude-newer=<DATE>` to re-lock the packages to versions before the specific date: https://pdm-project.org/en/latest/usage/lockfile/#exclude-packages-newer-than-specific-date
You need to run `pdm install` after the lock is done.

**Reply by dayongxie** [reply, marked as answer]
Thank you, I found a better way for me from the document: 
https://pdm-project.org/en/latest/usage/dependency/#via-cli-option
```bash
pdm lock --override constraints.txt
pdm sync
```
